File: LC55.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        m = 0
        n = len(nums)

        for i in range(n):
            if i > m:
                return False
            
            m = max(m, i + nums[i])
            if m >= n-1:
                return True
        
        return True

File: LC55.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int m = 0, n = nums.size();

        for(int i = 0; i < n; i++){
            if(i > m)
                return false;
            m = max(m,i + nums[i]);

            if(m >= n-1)
                return true;
        }

        return true;
    }
};
